File: Coding Challenges/Daily Challenge/2026/January/20260130-Valid-Pawn-Moves.js
function findPawnMoves(position) {
  const columns = "ABCDEFGH";
  const oldCoords = [
    columns.indexOf(position.slice(0,1)),
    Number(position.slice(1)) - 1
  ];
  const possibleMoves = oldCoords[1] == 1 ? [[0,1],[0,2]] : [[0,1]];
  let possiblePos = []

  for(let i = 0; i < possibleMoves.length; i++) {
    let move = possibleMoves[i];
    let newPos = [oldCoords[0] + move[0], oldCoords[1] + move[1]];

    if(newPos[0] >= 0 && newPos[0] < 8 && newPos[1] >= 0 && newPos[1] < 8) {
      possiblePos.push(columns[newPos[0]] + (newPos[1] + 1));      
    }
  }
  
  return possiblePos;
}

File: Coding Challenges/Daily Challenge/Python/20260130_ValidPawnMoves.py
def find_pawn_moves(position):    
    columns = "ABCDEFGH"
    old_coords = [
        columns.index(position[:1]),
        int(position[1:]) - 1
    ]
    possible_moves = [[0,1],[0,2]] if old_coords[1] == 1 else [[0,1]]
    possible_pos = []

    for move in possible_moves:
        new_pos = [old_coords[0] + move[0], old_coords[1] + move[1]]

        if new_pos[0] >= 0 and new_pos[0] < 8 and new_pos[1] >= 0 and new_pos[1] < 8:
            possible_pos.append(columns[new_pos[0]] + str(new_pos[1] + 1))


    return possible_pos